readme.md

```diff
@@ -1,53 +1,170 @@
-# Web API Skeleton
+#Catalog API Test
 
-Skeleton for Dropwizard Web APIs.
+##Overview
+A test implementation of RESTful API with courses scraped from the online catalog.
 
+##Description
+Catalog API test is a web app API allowing users to access information on various computer science courses at Oregon State University.  There are two main components to this API, the scraper and the actual RESTful API.
 
-## Tasks
+###Web Scraper
+Course information will be scraped from the online catalog and stored for use (most likely with a database).  The data will include course information such as the course name, description, instructor, CRN, day, time, and location.  In order to make the scraping process not too aggressive on the host server, only one disciplinary course (computer science) will be scraped, versus other scrapers which hit every single course offered.  The scraping process will be made runnable via an API call.
 
-List all tasks runnable from root project:
+###RESTful API
+The API will grab the data from the database and provide an outlet for users to request and interact with the data.  The API as a whole as well as the methods of interaction will be highlighted in the next section.
 
-    $ gradle tasks
+##Features
+Users will be able to access each course’s information through a RESTful interface.  With this data, they will be able to build any course related service.
 
-### IntelliJ IDEA
+With requests, users will be able to access each courses information as listed above.  They will be able to use GET to retrieve data from a resource and PUT to update a resource.  If invalid requests are submitted, ie. attempting to retrieve a non-existing resource or submitting a PUT request with invalid data, an error message will be outputted to the user - examples are as shown in the mockup section.
 
-Generate IntelliJ IDEA project:
+**Please note that if scraper is run again after any submitted PUT requests, those changes will be overwritten by the scraped data.
 
-    $ gradle idea
+Courses will be accessible by search through the course’s CRN.  For example CRN #50220 may reference one of the first computer science courses, ie. CS 101. And CRN #11933 may reference another course that follows, ie. CS 160.
 
-Open with `File` -> `Open Project`.
+### Setup Instructions
 
-### Build
+Open terminal, navigate to your cloned local directory, and enter the following line
```

Comment:
note that it's necessary to
1. `gradle build`, and
2. add credentials to configuration file
before you can run the application.